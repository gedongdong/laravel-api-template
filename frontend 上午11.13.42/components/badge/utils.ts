import { PresetColorTypes } from '../_util/colors';

export function isPresetColor(color?: string): boolean {
  return (PresetColorTypes as any[]).indexOf(color) !== -1;
}
