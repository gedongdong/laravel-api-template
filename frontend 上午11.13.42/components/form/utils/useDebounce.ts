import type { Ref } from 'vue';
import { shallowRef, watchEffect } from 'vue';

export default function useDebounce<T>(value: Ref<T[]>): Ref<T[]> {
  const cacheValue = shallowRef(value.value.slice());
  let timeout: any = null;
  watchEffect(() => {
    clearTimeout(timeout);
    timeout = setTimeout(
      () => {
        cacheValue.value = value.value;
      },
      value.value.length ? 0 : 10,
    );
  });

  return cacheValue;
}
