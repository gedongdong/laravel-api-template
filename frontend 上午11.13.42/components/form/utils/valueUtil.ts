import { toArray } from './typeUtil';
import type { InternalNamePath, NamePath } from '../interface';
import get from '../../vc-util/get';
import set from '../../vc-util/set';

/**
 * Convert name to internal supported format.
 * This function should keep since we still thinking if need support like `a.b.c` format.
 * 'a' => ['a']
 * 123 => [123]
 * ['a', 123] => ['a', 123]
 */
export function getNamePath(path: NamePath | null): InternalNamePath {
  return toArray(path);
}

export function getValue<T>(store: T, namePath: InternalNamePath) {
  const value = get(store, namePath);
  return value;
}

export function setValue<T>(
  store: T,
  namePath: InternalNamePath,
  value: any,
  removeIfUndefined = false,
): T {
  const newStore = set(store, namePath, value, removeIfUndefined);
  return newStore;
}

export function containsNamePath(namePathList: InternalNamePath[], namePath: InternalNamePath) {
  return namePathList && namePathList.some(path => matchNamePath(path, namePath));
}

function isObject(obj: any) {
  return typeof obj === 'object' && obj !== null && Object.getPrototypeOf(obj) === Object.prototype;
}

/**
 * Copy values into store and return a new values object
 * ({ a: 1, b: { c: 2 } }, { a: 4, b: { d: 5 } }) => { a: 4, b: { c: 2, d: 5 } }
 */
function internalSetValues<T>(store: T, values: T): T {
  const newStore: T = (Array.isArray(store) ? [...store] : { ...store }) as T;

  if (!values) {
    return newStore;
  }

  Object.keys(values).forEach(key => {
    const prevValue = newStore[key];
    const value = values[key];

    // If both are object (but target is not array), we use recursion to set deep value
    const recursive = isObject(prevValue) && isObject(value);
    newStore[key] = recursive ? internalSetValues(prevValue, value || {}) : value;
  });

  return newStore;
}

export function setValues<T>(store: T, ...restValues: T[]): T {
  return restValues.reduce(
    (current: T, newStore: T) => internalSetValues(current, newStore),
    store,
  );
}

export function cloneByNamePathList<T>(store: T, namePathList: InternalNamePath[]): T {
  let newStore = {} as T;
  namePathList.forEach(namePath => {
    const value = getValue(store, namePath);
    newStore = setValue(newStore, namePath, value);
  });

  return newStore;
}

export function matchNamePath(
  namePath: InternalNamePath,
  changedNamePath: InternalNamePath | null,
) {
  if (!namePath || !changedNamePath || namePath.length !== changedNamePath.length) {
    return false;
  }
  return namePath.every((nameUnit, i) => changedNamePath[i] === nameUnit);
}
