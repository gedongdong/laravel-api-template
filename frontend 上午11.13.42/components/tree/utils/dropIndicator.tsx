import type { CSSProperties } from 'vue';

export const offset = 4;

export default function dropIndicatorRender(props: {
  dropPosition: -1 | 0 | 1;
  dropLevelOffset: number;
  indent: number;
  prefixCls: string;
  direction: 'ltr' | 'rtl';
}) {
  const { dropPosition, dropLevelOffset, prefixCls, indent, direction = 'ltr' } = props;
  const startPosition = direction === 'ltr' ? 'left' : 'right';
  const endPosition = direction === 'ltr' ? 'right' : 'left';
  const style: CSSProperties = {
    [startPosition]: `${-dropLevelOffset * indent + offset}px`,
    [endPosition]: 0,
  };
  switch (dropPosition) {
    case -1:
      style.top = `${-3}px`;
      break;
    case 1:
      style.bottom = `${-3}px`;
      break;
    default:
      // dropPosition === 0
      style.bottom = `${-3}px`;
      style[startPosition] = `${indent + offset}px`;
      break;
  }
  return <div style={style} class={`${prefixCls}-drop-indicator`} />;
}
