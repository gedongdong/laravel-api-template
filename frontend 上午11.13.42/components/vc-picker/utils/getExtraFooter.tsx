import type { PanelMode } from '../interface';

export default function getExtraFooter(
  prefixCls: string,
  mode: PanelMode,
  renderExtraFooter?: (mode: PanelMode) => any,
) {
  if (!renderExtraFooter) {
    return null;
  }

  return <div class={`${prefixCls}-footer-extra`}>{renderExtraFooter(mode)}</div>;
}
