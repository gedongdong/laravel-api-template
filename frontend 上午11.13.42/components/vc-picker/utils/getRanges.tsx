import type { VueNode } from '../../_util/type';
import type { Components, RangeList, Locale } from '../interface';

export type RangesProps = {
  prefixCls: string;
  rangeList?: RangeList;
  components?: Components;
  needConfirmButton: boolean;
  onNow?: null | (() => void) | false;
  onOk?: null | (() => void) | false;
  okDisabled?: boolean;
  showNow?: boolean;
  locale: Locale;
};

export default function getRanges({
  prefixCls,
  rangeList = [],
  components = {},
  needConfirmButton,
  onNow,
  onOk,
  okDisabled,
  showNow,
  locale,
}: RangesProps) {
  let presetNode: VueNode;
  let okNode: VueNode;

  if (rangeList.length) {
    const Item = (components.rangeItem || 'span') as any;

    presetNode = (
      <>
        {rangeList.map(({ label, onClick, onMouseenter, onMouseleave }) => (
          <li key={label} class={`${prefixCls}-preset`}>
            <Item onClick={onClick} onMouseenter={onMouseenter} onMouseleave={onMouseleave}>
              {label}
            </Item>
          </li>
        ))}
      </>
    );
  }

  if (needConfirmButton) {
    const Button = (components.button || 'button') as any;

    if (onNow && !presetNode && showNow !== false) {
      presetNode = (
        <li class={`${prefixCls}-now`}>
          <a class={`${prefixCls}-now-btn`} onClick={onNow}>
            {locale.now}
          </a>
        </li>
      );
    }

    okNode = needConfirmButton && (
      <li class={`${prefixCls}-ok`}>
        <Button disabled={okDisabled} onClick={onOk}>
          {locale.ok}
        </Button>
      </li>
    );
  }

  if (!presetNode && !okNode) {
    return null;
  }

  return (
    <ul class={`${prefixCls}-ranges`}>
      {presetNode}
      {okNode}
    </ul>
  );
}
