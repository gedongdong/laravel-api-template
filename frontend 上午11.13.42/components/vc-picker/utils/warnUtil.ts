import { warning } from '../../vc-util/warning';
import type { DisabledTimes, PickerMode } from '../interface';

export interface WarningProps extends DisabledTimes {
  picker?: PickerMode;
}

export function legacyPropsWarning(props: WarningProps) {
  const { picker, disabledHours, disabledMinutes, disabledSeconds } = props;

  if (picker === 'time' && (disabledHours || disabledMinutes || disabledSeconds)) {
    warning(
      false,
      `'disabledHours', 'disabledMinutes', 'disabledSeconds' will be removed in the next major version, please use 'disabledTime' instead.`,
    );
  }
}
