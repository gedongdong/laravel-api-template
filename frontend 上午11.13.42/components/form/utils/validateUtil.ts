import RawAsyncValidator from 'async-validator';
import { cloneVNode } from 'vue';
import { warning } from '../../vc-util/warning';

import { setValues } from './valueUtil';
import { defaultValidateMessages } from './messages';
import { isValidElement } from '../../_util/props-util';
import type { InternalNamePath, RuleError, RuleObject, ValidateOptions } from '../interface';

// Remove incorrect original ts define
const AsyncValidator: any = RawAsyncValidator;

/**
 * Replace with template.
 *   `I'm ${name}` + { name: 'bamboo' } = I'm bamboo
 */
function replaceMessage(template: string, kv: Record<string, string>): string {
  return template.replace(/\$\{\w+\}/g, (str: string) => {
    const key = str.slice(2, -1);
    return kv[key];
  });
}

async function validateRule(
  name: string,
  value: any,
  rule: RuleObject,
  options: ValidateOptions,
  messageVariables?: Record<string, string>,
): Promise<string[]> {
  const cloneRule = { ...rule };

  // Bug of `async-validator`
  delete (cloneRule as any).ruleIndex;
  delete (cloneRule as any).trigger;

  // We should special handle array validate
  let subRuleField: RuleObject = null;
  if (cloneRule && cloneRule.type === 'array' && cloneRule.defaultField) {
    subRuleField = cloneRule.defaultField;
    delete cloneRule.defaultField;
  }

  const validator = new AsyncValidator({
    [name]: [cloneRule],
  });

  const messages = setValues({}, defaultValidateMessages, options.validateMessages);
  validator.messages(messages);

  let result = [];

  try {
    await Promise.resolve(validator.validate({ [name]: value }, { ...options }));
  } catch (errObj) {
    if (errObj.errors) {
      result = errObj.errors.map(({ message }, index: number) =>
        // Wrap VueNode with `key`
        isValidElement(message) ? cloneVNode(message, { key: `error_${index}` }) : message,
      );
    } else {
      console.error(errObj);
      result = [(messages.default as () => string)()];
    }
  }

  if (!result.length && subRuleField) {
    const subResults: string[][] = await Promise.all(
      (value as any[]).map((subValue: any, i: number) =>
        validateRule(`${name}.${i}`, subValue, subRuleField, options, messageVariables),
      ),
    );

    return subResults.reduce((prev, errors) => [...prev, ...errors], []);
  }

  // Replace message with variables
  const kv = {
    ...(rule as Record<string, string | number>),
    name,
    enum: (rule.enum || []).join(', '),
    ...messageVariables,
  };

  const fillVariableResult = result.map(error => {
    if (typeof error === 'string') {
      return replaceMessage(error, kv);
    }
    return error;
  });

  return fillVariableResult;
}

/**
 * We use `async-validator` to validate the value.
 * But only check one value in a time to avoid namePath validate issue.
 */
export function validateRules(
  namePath: InternalNamePath,
  value: any,
  rules: RuleObject[],
  options: ValidateOptions,
  validateFirst: boolean | 'parallel',
  messageVariables?: Record<string, string>,
) {
  const name = namePath.join('.');

  // Fill rule with context
  const filledRules: RuleObject[] = rules
    .map((currentRule, ruleIndex) => {
      const originValidatorFunc = currentRule.validator;
      const cloneRule = {
        ...currentRule,
        ruleIndex,
      };

      // Replace validator if needed
      if (originValidatorFunc) {
        cloneRule.validator = (rule: RuleObject, val: any, callback: (error?: string) => void) => {
          let hasPromise = false;

          // Wrap callback only accept when promise not provided
          const wrappedCallback = (...args: string[]) => {
            // Wait a tick to make sure return type is a promise
            Promise.resolve().then(() => {
              warning(
                !hasPromise,
                'Your validator function has already return a promise. `callback` will be ignored.',
              );

              if (!hasPromise) {
                callback(...args);
              }
            });
          };

          // Get promise
          const promise = originValidatorFunc(rule, val, wrappedCallback);
          hasPromise =
            promise && typeof promise.then === 'function' && typeof promise.catch === 'function';

          /**
           * 1. Use promise as the first priority.
           * 2. If promise not exist, use callback with warning instead
           */
          warning(hasPromise, '`callback` is deprecated. Please return a promise instead.');

          if (hasPromise) {
            (promise as Promise<void>)
              .then(() => {
                callback();
              })
              .catch(err => {
                callback(err || ' ');
              });
          }
        };
      }

      return cloneRule;
    })
    .sort(({ warningOnly: w1, ruleIndex: i1 }, { warningOnly: w2, ruleIndex: i2 }) => {
      if (!!w1 === !!w2) {
        // Let keep origin order
        return i1 - i2;
      }

      if (w1) {
        return 1;
      }

      return -1;
    });

  // Do validate rules
  let summaryPromise: Promise<RuleError[]>;

  if (validateFirst === true) {
    // >>>>> Validate by serialization
    summaryPromise = new Promise(async (resolve, reject) => {
      /* eslint-disable no-await-in-loop */
      for (let i = 0; i < filledRules.length; i += 1) {
        const rule = filledRules[i];
        const errors = await validateRule(name, value, rule, options, messageVariables);
        if (errors.length) {
          reject([{ errors, rule }]);
          return;
        }
      }
      /* eslint-enable */

      resolve([]);
    });
  } else {
    // >>>>> Validate by parallel
    const rulePromises: Promise<RuleError>[] = filledRules.map(rule =>
      validateRule(name, value, rule, options, messageVariables).then(errors => ({ errors, rule })),
    );

    summaryPromise = (
      validateFirst ? finishOnFirstFailed(rulePromises) : finishOnAllFailed(rulePromises)
    ).then((errors: RuleError[]): RuleError[] | Promise<RuleError[]> => {
      // Always change to rejection for Field to catch
      return Promise.reject<RuleError[]>(errors);
    });
  }

  // Internal catch error to avoid console error log.
  summaryPromise.catch(e => e);

  return summaryPromise;
}

async function finishOnAllFailed(rulePromises: Promise<RuleError>[]): Promise<RuleError[]> {
  return Promise.all(rulePromises).then(
    (errorsList: RuleError[]): RuleError[] | Promise<RuleError[]> => {
      const errors: RuleError[] = [].concat(...errorsList);

      return errors;
    },
  );
}

async function finishOnFirstFailed(rulePromises: Promise<RuleError>[]): Promise<RuleError[]> {
  let count = 0;

  return new Promise(resolve => {
    rulePromises.forEach(promise => {
      promise.then(ruleError => {
        if (ruleError.errors.length) {
          resolve([ruleError]);
        }

        count += 1;
        if (count === rulePromises.length) {
          resolve([]);
        }
      });
    });
  });
}
