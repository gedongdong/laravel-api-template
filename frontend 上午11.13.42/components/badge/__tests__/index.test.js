import { mount } from '@vue/test-utils';
import Badge from '../index';
import mountTest from '../../../tests/shared/mountTest';

import { asyncExpect } from '../../../tests/utils';
describe('Badge', () => {
  it('badge dot not scaling count > 9', () => {
    const badge = mount({
      render() {
        return <Badge count={10} dot />;
      },
    });
    expect(badge.findAll('.ant-card-multiple-words').length).toBe(0);
  });
  it('badge should support float number', () => {
    let wrapper = mount({
      render() {
        return <Badge count={3.5} />;
      },
    });
    expect(wrapper.text()).toMatchSnapshot();
    wrapper = mount({
      render() {
        return <Badge count={3.5} />;
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('badge dot not showing count == 0', () => {
    const badge = mount({
      render() {
        return <Badge count={0} dot />;
      },
    });
    expect(badge.findAll('.ant-badge-dot').length).toBe(0);
  });

  it('should have an overriden title attribute', () => {
    const badge = mount({
      render() {
        return <Badge count={10} title="Custom title" />;
      },
    });
    expect(badge.find('.ant-scroll-number').element.attributes.getNamedItem('title').value).toEqual(
      'Custom title',
    );
  });

  // https://github.com/ant-design/ant-design/issues/10626
  // it('should be composable with Tooltip', async () => {
  //   const wrapper = mount({
  //     render () {
  //       return <Tooltip ref='tooltip' title='Fix the error'>
  //         <Badge status='error' />
  //       </Tooltip>
  //     },
  //   }, { sync: false })
  //   await asyncExpect(() => {
  //     wrapper.find({ name: 'ABadge' }).trigger('mouseenter')
  //   }, 0)

  //   expect(wrapper.vm.$refs.tooltip.sVisible).toBe(true)
  // })

  it('should render when count is changed', async () => {
    const wrapper = mount(Badge, {
      props: {
        count: 9,
      },
      sync: false,
    });
    await asyncExpect(() => {
      wrapper.setProps({ count: 10 });
    }, 100);
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
      wrapper.setProps({ count: 11 });
    }, 100);
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
      wrapper.setProps({ count: 11 });
    }, 100);
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
      wrapper.setProps({ count: 10 });
    }, 100);
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
      wrapper.setProps({ count: 9 });
    }, 100);
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
    }, 100);
  });

  it('should be compatible with borderColor style', () => {
    const wrapper = mount({
      render() {
        return (
          <Badge
            count={4}
            style={{ backgroundColor: '#fff', color: '#999', borderColor: '#d9d9d9' }}
          />
        );
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // https://github.com/ant-design/ant-design/issues/13694
  it('should support offset when count is a VueNode', () => {
    const wrapper = mount({
      render() {
        return (
          <Badge count={<span class="custom" style={{ color: '#f5222d' }} />} offset={[10, 20]}>
            <a href="#" class="head-example">
              head
            </a>
          </Badge>
        );
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  // https://github.com/ant-design/ant-design/issues/15799
  it('render correct with negative number', () => {
    const wrapper = mount({
      render() {
        return (
          <div>
            <Badge count="-10" />
            <Badge count={-10} />
          </div>
        );
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('text works with vnode', () => {
    const wrapper = mount({
      render() {
        return <Badge status="success" text={<span>hello</span>} />;
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('Ribbon', () => {
  mountTest(Badge.Ribbon);

  describe('placement', () => {
    it('works with `start` & `end` placement', () => {
      const wrapperStart = mount({
        render() {
          return (
            <Badge.Ribbon placement="start">
              <div />
            </Badge.Ribbon>
          );
        },
      });

      expect(wrapperStart.findAll('.ant-ribbon-placement-start').length).toEqual(1);

      const wrapperEnd = mount({
        render() {
          return (
            <Badge.Ribbon placement="end">
              <div />
            </Badge.Ribbon>
          );
        },
      });
      expect(wrapperEnd.findAll('.ant-ribbon-placement-end').length).toEqual(1);
    });
  });

  describe('color', () => {
    it('works with preset color', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge.Ribbon color="green">
              <div />
            </Badge.Ribbon>
          );
        },
      });
      expect(wrapper.findAll('.ant-ribbon-color-green').length).toEqual(1);
    });
  });

  describe('text', () => {
    it('works with string', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge.Ribbon text="cool">
              <div />
            </Badge.Ribbon>
          );
        },
      });
      expect(wrapper.find('.ant-ribbon').text()).toEqual('cool');
    });
    it('works with element', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge.Ribbon text={<span class="cool" />}>
              <div />
            </Badge.Ribbon>
          );
        },
      });
      expect(wrapper.findAll('.cool').length).toEqual(1);
    });
  });
});
