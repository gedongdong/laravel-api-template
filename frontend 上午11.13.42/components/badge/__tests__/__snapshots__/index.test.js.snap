// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge badge should support float number 1`] = `"3.5"`;

exports[`Badge badge should support float number 2`] = `<span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="3.5">3.5</sup><!----></span>`;

exports[`Badge render correct with negative number 1`] = `
<div><span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="-10"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">-</p></span><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">1</p></span><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">0</p></span></sup>
  <!----></span><span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="-10"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">-</p></span><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">1</p></span><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">0</p></span></sup>
  <!----></span>
</div>
`;

exports[`Badge should be compatible with borderColor style 1`] = `
<span style="background-color: rgb(255, 255, 255); color: rgb(153, 153, 153); border-color: #d9d9d9;" class="ant-badge ant-badge-not-a-wrapper"><!----><sup style="background-color: rgb(255, 255, 255); color: rgb(153, 153, 153); border-color: #d9d9d9; box-shadow: 0 0 0 1px #d9d9d9 inset;" data-show="true" class="ant-scroll-number ant-badge-count" title="4"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">4</p></span></sup>
<!----></span>
`;

exports[`Badge should render when count is changed 1`] = `
<span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="10"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">1</p></span><span class="ant-scroll-number-only" style="transform: translateY(-100%);"><p class="ant-scroll-number-only-unit" style="position: absolute; top: -900%; left: 0px;">0</p><p style="position: absolute; top: -800%; left: 0px;" class="ant-scroll-number-only-unit">1</p><p style="position: absolute; top: -700%; left: 0px;" class="ant-scroll-number-only-unit">2</p><p style="position: absolute; top: -600%; left: 0px;" class="ant-scroll-number-only-unit">3</p><p style="position: absolute; top: -500%; left: 0px;" class="ant-scroll-number-only-unit">4</p><p style="position: absolute; top: -400%; left: 0px;" class="ant-scroll-number-only-unit">5</p><p style="position: absolute; top: -300%; left: 0px;" class="ant-scroll-number-only-unit">6</p><p style="position: absolute; top: -200%; left: 0px;" class="ant-scroll-number-only-unit">7</p><p style="position: absolute; top: -100%; left: 0px;" class="ant-scroll-number-only-unit">8</p><p class="ant-scroll-number-only-unit current">9</p><p style="position: absolute; top: 100%; left: 0px;" class="ant-scroll-number-only-unit">0</p></span></sup>
<!----></span>
`;

exports[`Badge should render when count is changed 2`] = `
<span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="11"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">1</p></span><span class="ant-scroll-number-only" style="transform: translateY(-200%);"><p class="ant-scroll-number-only-unit" style="position: absolute; top: -800%; left: 0px;">1</p><p style="position: absolute; top: -700%; left: 0px;" class="ant-scroll-number-only-unit">2</p><p style="position: absolute; top: -600%; left: 0px;" class="ant-scroll-number-only-unit">3</p><p style="position: absolute; top: -500%; left: 0px;" class="ant-scroll-number-only-unit">4</p><p style="position: absolute; top: -400%; left: 0px;" class="ant-scroll-number-only-unit">5</p><p style="position: absolute; top: -300%; left: 0px;" class="ant-scroll-number-only-unit">6</p><p style="position: absolute; top: -200%; left: 0px;" class="ant-scroll-number-only-unit">7</p><p style="position: absolute; top: -100%; left: 0px;" class="ant-scroll-number-only-unit">8</p><p class="ant-scroll-number-only-unit current">9</p><p class="ant-scroll-number-only-unit" style="position: absolute; top: 100%; left: 0px;">0</p><p style="position: absolute; top: 200%; left: 0px;" class="ant-scroll-number-only-unit">1</p></span></sup>
<!----></span>
`;

exports[`Badge should render when count is changed 3`] = `
<span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="11"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">1</p></span><span class="ant-scroll-number-only" style="transform: translateY(-200%);"><p class="ant-scroll-number-only-unit" style="position: absolute; top: -800%; left: 0px;">1</p><p style="position: absolute; top: -700%; left: 0px;" class="ant-scroll-number-only-unit">2</p><p style="position: absolute; top: -600%; left: 0px;" class="ant-scroll-number-only-unit">3</p><p style="position: absolute; top: -500%; left: 0px;" class="ant-scroll-number-only-unit">4</p><p style="position: absolute; top: -400%; left: 0px;" class="ant-scroll-number-only-unit">5</p><p style="position: absolute; top: -300%; left: 0px;" class="ant-scroll-number-only-unit">6</p><p style="position: absolute; top: -200%; left: 0px;" class="ant-scroll-number-only-unit">7</p><p style="position: absolute; top: -100%; left: 0px;" class="ant-scroll-number-only-unit">8</p><p class="ant-scroll-number-only-unit current">9</p><p class="ant-scroll-number-only-unit" style="position: absolute; top: 100%; left: 0px;">0</p><p style="position: absolute; top: 200%; left: 0px;" class="ant-scroll-number-only-unit">1</p></span></sup>
<!----></span>
`;

exports[`Badge should render when count is changed 4`] = `
<span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count ant-badge-multiple-words" title="10"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">1</p></span><span class="ant-scroll-number-only" style="transform: translateY(-100%);"><p class="ant-scroll-number-only-unit" style="position: absolute; top: -900%; left: 0px;">0</p><p style="position: absolute; top: -800%; left: 0px;" class="ant-scroll-number-only-unit">1</p><p style="position: absolute; top: -700%; left: 0px;" class="ant-scroll-number-only-unit">2</p><p style="position: absolute; top: -600%; left: 0px;" class="ant-scroll-number-only-unit">3</p><p style="position: absolute; top: -500%; left: 0px;" class="ant-scroll-number-only-unit">4</p><p style="position: absolute; top: -400%; left: 0px;" class="ant-scroll-number-only-unit">5</p><p style="position: absolute; top: -300%; left: 0px;" class="ant-scroll-number-only-unit">6</p><p style="position: absolute; top: -200%; left: 0px;" class="ant-scroll-number-only-unit">7</p><p class="ant-scroll-number-only-unit" style="position: absolute; top: -100%; left: 0px;">8</p><p class="ant-scroll-number-only-unit current">9</p><p style="position: absolute; top: 100%; left: 0px;" class="ant-scroll-number-only-unit">0</p></span></sup>
<!----></span>
`;

exports[`Badge should render when count is changed 5`] = `
<span class="ant-badge ant-badge-not-a-wrapper"><!----><sup data-show="true" class="ant-scroll-number ant-badge-count" title="9"><span class="ant-scroll-number-only" style="transition: none;"><p class="ant-scroll-number-only-unit current">9</p></span></sup>
<!----></span>
`;

exports[`Badge should support offset when count is a VueNode 1`] = `
<span class="ant-badge"><a href="#" class="head-example">head</a><span class="custom ant-scroll-number-custom-component" style="color: rgb(245, 34, 45); margin-top: 20px; right: -10px;"></span>
<!----></span>
`;

exports[`Badge text works with vnode 1`] = `<span class="ant-badge ant-badge-status ant-badge-not-a-wrapper"><span class="ant-badge-status-dot ant-badge-status-success"></span><span class="ant-badge-status-text"><span>hello</span></span></span>`;
